tests/main/snapd-apparmor/task.yaml

```diff
@@ -0,0 +1,73 @@
+summary: Ensure snapd-apparmor works as expected
+
+environment:
+    CONSUMER_SNAP: test-snapd-policy-app-consumer
+
+debug: |
+    journalctl -u snap.apparmor.service
+
+execute: |
+    if ! systemctl is-active snapd.apparmor.service; then
+      echo "Skipping test since snapd.apparmor.service is not active"
+      exit 0
+    fi
+
+    echo "Ensure snapd.apparmor is enabled"
+    systemctl is-enabled snapd.apparmor.service
+
+    # install a test snap which generates a lot of apparmor policies
+    echo "Given a test snap is installed"
+    "$TESTSTOOLS"/snaps-state install-local "$CONSUMER_SNAP"
+    tests.cleanup defer snap remove --purge "$CONSUMER_SNAP"
+
+    # stop snapd so it does not try and load apparmor policies directly
+    echo "And snapd is stopped"
+    systemctl stop snapd.service
+    tests.cleanup defer systemctl restart snapd.service
+
+    # get the current set of profiles but ignore any which may have gotten
+    # created on-the-fly by apparmor for namespaced commands - these
+    # contain / - as these won't get automatically recreated by
+    # snapd.apparmor
+    grep -v / /sys/kernel/security/apparmor/profiles | cut -f1 -d" " | sort > profiles.txt
+
+    # manually unload all profiles defined by snap apparmor policies
+    echo "If we unload existing snap apparmor policy"
+    for p in /var/lib/snapd/apparmor/profiles/*; do
+      while IFS= read -r profile; do
+        echo "Unloading $profile..."
```

Comment:
should we first check that it is loaded? Maybe using the same regexp that we use with NOMATCH below, so that we verify that it is correct.

Reply:
We could try and test first, but if the profile is not loaded and we just go ahead and unload it, then the command fails and so the whole test will fail, which I think is a good thing, as then it can test the assumption that all the profiles are already loaded. Whereas if we only unload the profile when we know it is already loaded then we will not catch the case where profiles have failed to be loaded beforehand. So I would prefer to keep it as is.

Reply:
Ah, I didn't know taht the command would fail if we attempted to unload a profile which is not loaded. Then indeed it's fine as it is.